src/source/shift.rs

```diff
@@ -485,8 +486,39 @@ impl FromShift {
             Some("IdentifierExpression") => {
                 debug!(target: "Shift", "FromShift IdentifierExpression {:?}", object);
             }
+            Some("IfStatement") => {
```

Comment:
Hmm could this check be moved to use the resulting typed format? I'm trying to get rid of an intermediate JSON AST, and, in particular, started with moving these conversions in a parallel PR.